buildSrc/src/main/kotlin/ExecutableJar.kt

```diff
@@ -15,27 +15,25 @@ import org.gradle.kotlin.dsl.listProperty
  *
  * https://skife.org/java/unix/2011/06/20/really_executable_jars.html
  */
-open class ExecutableJar : DefaultTask() {
+abstract class ExecutableJar : DefaultTask() {
   @get:InputFile
-  val inJar: RegularFileProperty = project.objects.fileProperty()
+  abstract val inJar: RegularFileProperty
 
   @get:OutputFile
-  val outJar: RegularFileProperty = project.objects.fileProperty()
+  abstract val outJar: RegularFileProperty
 
   @get:Input
-  val jvmArgs: ListProperty<String> = project.objects.listProperty()
+  abstract val jvmArgs: ListProperty<String>
```

Comment:
This change is unrelated to Windows. Happy to move this to a separate PR if you'd prefer.